### PR TITLE
Backports/708/v3

### PR DESCRIPTION
**Pull request by victorjulien**
#12263, with https://github.com/OISF/suricata/pull/12266/commits/5d22aa4d4b40ae60e442fd40e76611c2f8c17ff0

### CODE REVIEW

**Review by catenacyber — APPROVED** [review]
Original PR ok for me, plus added doc commit ok for me

File: doc/userguide/output/eve/eve-json-output.rst [hunk 1]
```diff
@@ -71,11 +71,11 @@ can be used to force the detect engine to tie a transaction
...
-If there are multiple live transactions, none will get
-picked up.
-The alert event will have ``"tx_guessed": true`` to recognize
-these alerts.
-
+**WARNING: If there are multiple live transactions, none will get
+picked up.** This is to reduce the chances of logging unrelated data, and may
+lead to alerts being logged without metadata, in some cases.
+The alert event will have ``tx_guessed: true`` to recognize
+such alerts.
```
Comment (catenacyber) [inline doc/userguide/output/eve/eve-json-output.rst:78]: Do you want to add there that we also log some transaction metadata for rule with stream matches ?
And that the first transaction is logged, like one TCP packet with 3 DNS requests to suricata.io  oisf.net and suricon.net and rule `alert tcp any any -> any any (content: "suricon"; sid: 1)` will log the request to suricata.io

File: suricata.yaml.in [hunk 2]
```diff
@@ -1677,6 +1677,11 @@ detect:
     toserver-groups: 25
   sgh-mpm-context: auto
   inspection-recursion-limit: 3000
+  # try to tie an app-layer transaction for rules without app-layer keywords
+  # if there is only one live transaction for the flow
+  # allows to log app-layer metadata in alert
+  # but the transaction may not be the relevant one.
```
Comment (jufajardini) [inline suricata.yaml.in:1683]: Tried to add a bit more here (from https://github.com/OISF/suricata/pull/12260/files):
```suggestion
  # Try to guess an app-layer transaction for rules without app-layer keywords,
  # ONLY IF there is just one live transaction for the flow.
  # This allows logging app-layer metadata in alert - the transaction may not
  # be the relevant one for the alert.
```

**Comment by jasonish** [comment]
Should pass with:

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2179

File: doc/userguide/output/eve/eve-json-output.rst (hunk 1 above)
Comment (jasonish) [reply doc/userguide/output/eve/eve-json-output.rst:78]: Seems all too much of a low level implementation detail to discuss in configuration docs.  But these types of issues are making me think we need a know issues and shortcomings chapter.

File: suricata.yaml.in (hunk 2 above)
Comment (jasonish) [reply suricata.yaml.in:1683]: > Tried to add a bit more here (from https://github.com/OISF/suricata/pull/12260/files):

On suggestion...

> ONLY IF there is a single live transaction for the flow,

IMO, can be cleaned up in post-release.

**Review by jasonish — APPROVED** [review]
Stage with #12264 and https://github.com/OISF/suricata-verify/pull/2179

**Comment by victorjulien** [comment]
replaced by #12268

**Pull request closed without merging**